compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1255,20 +1256,24 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
                                     continue;
                                 }
                                 this.insert_lifetime(lt, Region::Static);
-                                this.tcx
-                                    .sess
-                                    .struct_span_warn(
-                                        lifetime.span,
-                                        &format!(
+                                this.tcx.struct_span_lint_hir(
+                                    lint::builtin::NAMED_STATIC_LIFETIMES,
+                                    lifetime.hir_id,
+                                    lifetime.span,
+                                    |lint| {
+                                        let msg = &format!(
                                             "unnecessary lifetime parameter `{}`",
                                             lifetime.name.ident(),
-                                        ),
-                                    )
-                                    .help(&format!(
-                                        "you can use the `'static` lifetime directly, in place of `{}`",
-                                        lifetime.name.ident(),
-                                    ))
-                                    .emit();
+                                        );
+                                        let help = &format!(
+                                            "you can use the `'static` lifetime directly, in place of `{}`",
+                                            lifetime.name.ident(),
+                                        );
+                                        lint.build(msg)
+                                            .help(help)
```

Comment:
Not necessarily something to do on this PR, but we have enough context to turn the `help` into a structured suggestion, by inspecting the bounds and suggesting:

```
LL -     type Y<'a: 'static>;
LL +     type Y<'static>;
   |            ~~~~~~~
```